test/utils.py

```diff
@@ -0,0 +1,32 @@
+import sys
+import yaml
+import json
+
+def add_sidecar_to_yaml_file():
+    # This function is expecting a tekton task from stdin
+    with sys.stdin as file:
+        # Load yaml file passed by stdin
+        payload = yaml.load(file, Loader=yaml.FullLoader)
+        # If sidecars doesnt have any entries create an array for payload manipulation
+        if 'sidecars' not in payload['spec']:
+            payload['spec']['sidecars'] = []
+        # Append the local image registry to the yaml file
+        payload['spec']['sidecars'].append({"image": "registry", "name": "registry"})
```

Comment:
Can we just specify the registry in the original YAML, instead of injecting in this way?

Reply:
I agree that injecting sidecars is not a good idea. I also think it would be a mistake to change the yaml for the sake of passing a test. The local image registry was only there to help us test `docker push`. Now, honestly its hurting.  I would perfer to push to my docker hub account then add an ephemeral image registry that provides no purpose for the end-user. Also, there are about 13 other tasks that use the add registry sidecar. Ignoring this method in docker-build does not fix the overall problem that this function does perform it's expected behavior. 
```bash
    cat ${TMPF}.read | python3 -c 'import yaml;f=open(0, encoding="utf-8"); data=yaml.load(f.read(), Loader=yaml.FullLoader);data["spec"]["sidecars"]=[{"image":"registry", "name": "registry"}];print(yaml.dump(data, default_flow_style=False));' > ${TMPF}
```
I do think having a namespace with a local image registry would be a good solution. Openshift provides an image registry that can be accessed in its own namespace. I think creating an uninsecured registry in its own namespace that gets cleaned up at the end of testing would be useful. Pods would be able to access it through it's svc. This would also help CI times cause we wouldn't have to spin up 14 image registry pods.

Reply:
Oh I see what you're saying, the registry shouldn't be specified in the actual catalog task. That definitely makes sense, I wasn't paying enough attention. 🤦 

+1 to having e2e tests start by running a registry, run all the relevant tasks against it, and then delete the registry. This would let us test the tasks without modifying them, which might also hide breakages.

Reply:
The only thing left that needs to be decided is a standard for naming and tagging images with a shared registry. (I.E we cant have 10 tasks build and push foo:latests to the registry without breaking the test).

Reply:
Since, tasks already have a version naming and tagging it with those labels would work.
example:
<task-name>:<task-version>
localhost:5000/docker-build:0.1